### ISSUE TITLE
Default zoom level is ignored on start by backend webengine

### ISSUE THREAD

**Issue by zuglufttier** [The-Compiler/qutebrowser#1993]
When I set the default zoom level via `:set ui default-zoom 150%` everything works as expected. However, newly opened tabs don't respect that setting and after a restart everything is back to 100% although the value of the zoom-level is saved in the config.

Increasing the zoom-level via `<ctrl>-+` leads to a zoom-level of 175%. The normal backend is working fine.

**Version Info**

qutebrowser v0.8.1
Git commit: v0.8.0-1336-g80b5c91 (2016-09-29 10:59:32 +0200)

CPython: 3.5.2
Qt: 5.7.0
PyQt: 5.7

sip: 4.18.1
colorama: no
pypeg2: 2.15
jinja2: 2.8
pygments: 2.1.3
yaml: 3.12
cssutils: no
typing: yes
PyQt5.QtWebEngineWidgets: yes
pdf.js: 1.4.20 (/usr/share/pdf.js/build/pdf.js)
Webkit: 538.1
SSL: OpenSSL 1.0.2j  26 Sep 2016

Style: Breeze::Style
Platform: Linux-4.7.4-1-ARCH-x86_64-with-arch, 64bit
Frozen: False
Imported from /usr/lib/python3.5/site-packages/qutebrowser
OS Version: 

--- /etc/os-release ---
NAME="Arch Linux"
ID=arch
PRETTY_NAME="Arch Linux"

Paths:
config: /home/zlt/.config/qutebrowser
system_data: /home/zlt/.local/share/qutebrowser
data: /home/zlt/.local/share/qutebrowser
download: /home/zlt/Downloads
cache: /home/zlt/.cache/qutebrowser
runtime: /run/user/1000/qutebrowser

**Comment by zuglufttier** [comment]
I think this is related to my other issue! KDE 5 let's the user choose a scaling factor of 1.5 in the GUI which results in this:

`~ % env | grep QT
QT_AUTO_SCREEN_SCALE_FACTOR=0
QT_SCREEN_SCALE_FACTORS=eDP1=1.5;DP1=1.5;DP2=1.5;HDMI1=1.5;HDMI2=1.5;VIRTUAL1=1.5;
`

Right now I can't seem to replicate this behaviour but this might just explain it. I'll have a look into this...

**Comment by The-Compiler** [comment]
So you can't reproduce the issue with an integer as `QT_SCREEN_SCALE_FACTORS`?

**Comment by zuglufttier** [comment]
I can reproduce it using an integer; I just double checked. I also set the scaling factor in the KDE GUI to 1 and restarted the system.

The issue does not seem to be related to `QT_SCREEN_SCALE_FACTORS`.

**Comment by lluixhi** [comment]
@The-Compiler I have the same issue here.
In my config file I have:
```
default-zoom = 125%
```
And at startup, it sets default zoom to 100%.
Once I issue the command
```
:set ui default-zoom 125%
```
The default zoom is set.

I'm using PyQt-5.7.0 and Qt-5.6.1 with qutebrowser:master on Gentoo.

**Comment by lluixhi** [comment]
Interestingly, pressing `=` after loading a page will set it to the default-zoom value.

**Comment by ml-** [comment]
Same issue. 
Using `default-zoom = 175%`

**"Finding the cause"** is on my local Task-list. I'll try to track this down when I have some free time between other stuff.

This is currently one of the main-reasons I'm still roling with the Webkit-backend.

**Comment by The-Compiler** [comment]
From IRC:

```
21:21 <chobojordi> anything I could do to help https://github.com/The-Compiler/qutebrowser/issues/1993 get fixed?  What I've seen is that webkittab.py has a self.zoom.set_default() call that is missing in webenginetab.py. But that's not fixing it
21:23 <chobojordi> or actually I think if I add such a call it is working for the fist page load in a new tab.. but as soon as I click on a link it will go back to 100% zoom
21:28 <chobojordi> once I've set the zoom with =  it stays the same within a tab
21:35 <chobojordi> ehm think I was wrong
21:35 <chobojordi> If I add self.zoom.set_default() any new tab has the correct zoom and as long as I only click on links of the same page it stays the same
21:36 <chobojordi> but if I open another url it resets to 100%
```

**Comment by lluixhi** [comment]
Yeah, same symptoms as chobojordi.

In addition, loading a session opens tabs with the correct default-zoom value.

**Comment by JonathanReeve** [comment]
I can confirm this issue, and I'd like to add that this makes Qutebrowser almost unusable on a HiDPI system, unless you're willing to press `=` every time you load a page.

**Comment by The-Compiler** [comment]
FWIW I'll soon get a HiDPI system, so after my exams are over (mid-January) I'll basically be forced to look at this in detail. No guarantees before that I'm afraid, because exams and stuff.

**Comment by The-Compiler** [comment]
The zoom is now set correctly initially, but due to [QTBUG-51851](https://bugreports.qt.io/browse/QTBUG-51851) I don't know what to do about subsequent pages being opened yet... I guess the best I can do at the moment is setting the zoom level when the page finished loading :unamused: 

However, I'll also experiment with telling Qt to turn on HighDPI scaling, which will hopefully scale things correctly without having to mess with `default-zoom`.

**Comment by The-Compiler** [comment]
So I just enabled `Qt::AA_EnableHighDpiScaling` in git master - I can't really test it yet (my highdpi laptop will finally arrive this or next week though, I hope...), but I'd love some feedback!

This should be equivalent to setting `QT_AUTO_SCREEN_SCALE_FACTOR=1` FWIW and should at least scale stuff up with QtWebEngine.

I also added a (slightly ugly) way of copying the zoom level when opening a new page.
